ibm/service/dnsservices/resource_ibm_private_dns_custom_resolver_secondary_zone.go

```diff
@@ -0,0 +1,269 @@
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+
+			pdnsSecondaryZoneEnabled: {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "Enable/Disable the secondary zone",
+			},
+
+			pdnsSecondaryZoneDescription: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Descriptive text of the secondary zone",
+			},
+
+			pdnsSecondaryZoneCreatedOn: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Secondary Zone Creation date",
+			},
+
+			pdnsSecondaryZoneModifiedOn: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Secondary Zone Modification date",
+			},
+		},
+	}
+}
+
+func resourceIBMPrivateDNSSecondaryZoneCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).PrivateDNSClientSession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	instanceID := d.Get(pdnsInstanceID).(string)
+	resolverID := d.Get(pdnsResolverID).(string)
+	description := d.Get(pdnsSecondaryZoneDescription).(string)
+	zone := d.Get(pdnsSecondaryZoneZone).(string)
+	enabled := d.Get(pdnsSecondaryZoneEnabled).(bool)
+	transferFrom := flex.ExpandStringList(d.Get(pdnsSecondaryZoneTransferFrom).([]interface{}))
+
+	createSecondaryZoneOptions := sess.NewCreateSecondaryZoneOptions(instanceID, resolverID)
+
+	createSecondaryZoneOptions.SetZone(zone)
+	createSecondaryZoneOptions.SetDescription(description)
+	createSecondaryZoneOptions.SetEnabled(enabled)
+	createSecondaryZoneOptions.SetTransferFrom(transferFrom)
+
+	mk := "private_dns_secondary_zone_" + instanceID + resolverID
+	conns.IbmMutexKV.Lock(mk)
+	defer conns.IbmMutexKV.Unlock(mk)
+
+	resource, response, err := sess.CreateSecondaryZone(createSecondaryZoneOptions)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("[ERROR] Error creating DNS Services secondary zone:%s\n%s", err, response))
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s/%s", instanceID, resolverID, *resource.ID))
+	return resourceIBMPrivateDNSSecondaryZoneRead(ctx, d, meta)
+}
+
+func resourceIBMPrivateDNSSecondaryZoneRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).PrivateDNSClientSession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	idSet := strings.Split(d.Id(), "/")
+	if len(idSet) < 3 {
+		return diag.FromErr(fmt.Errorf("[ERROR] Incorrect ID %s: Id should be a combination of InstanceID/resolverID/secondaryZoneID", d.Id()))
+	}
+	instanceID := idSet[0]
+	resolverID := idSet[1]
+	secondaryZoneID := idSet[2]
+	getSecondaryZoneOptions := sess.NewGetSecondaryZoneOptions(instanceID, resolverID, secondaryZoneID)
+	resource, response, err := sess.GetSecondaryZone(getSecondaryZoneOptions)
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("[ERROR] Error reading DNS Services secondary zone:%s\n%s", err, response))
```

Comment:
Can you check if 

Just add above return (this is a new way of writing code without Exists..Exists was used in old resources)
```
if response != nil && response.StatusCode == 404 {
			d.SetId("")
			return nil
			
		}
		```
		And you can remove the this line Exists:        resourceIBMPrivateDNSSecondaryZoneExists,